aptos-move/block-executor/src/executor.rs

```diff
@@ -623,11 +606,60 @@ where
             // TODO: re-use the code cache.
             drop(versioned_cache);
         });
+        ret
+    }
 
+    fn make_final_result_in_parallel(
+        &self,
+        last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
+        num_txns: usize,
+    ) -> Result<Vec<E::Output>, E::Error> {
+        let _timer = PARALLEL_EXECUTOR_FINAL_RESULT_SECONDS.start_timer();
+        let skip_rest_index = Arc::new(Mutex::new(None));
+        let maybe_err = Arc::new(Mutex::new(None));
+        let mut final_results = if last_input_output.module_publishing_may_race() {
+            counters::MODULE_PUBLISHING_FALLBACK_COUNT.inc();
+            *maybe_err.lock().unwrap() = Some(Error::ModulePathReadWrite);
+            vec![]
+        } else {
+            self.executor_thread_pool.scope(|_| {
+                (0..num_txns)
+                    .into_par_iter()
+                    .map(
+                        |idx| match last_input_output.try_take_output(idx as TxnIndex) {
+                            Some(ExecutionStatus::Success(output)) => output,
+                            Some(ExecutionStatus::SkipRest(output)) => {
+                                let mut skip_index = skip_rest_index.lock().unwrap();
+                                if let Some(existing_index) = *skip_index {
+                                    *skip_index = Some(std::cmp::min(existing_index, idx));
+                                } else {
+                                    *skip_index = Some(idx);
+                                }
+                                output
+                            },
+                            Some(ExecutionStatus::Abort(err)) => {
+                                *maybe_err.lock().unwrap() = Some(err);
```

Comment:
It's guaranteed that we do this once? right? maybe assert in debug mode?

Reply:
I don't think this is guaranteed.. @gelash can clarify...

Reply:
yeah, unfortunately atm everything gets processed, and then this part made sure it was the same as sequential (i.e. first skip took effect), also likely more errors could appear. I even remember something like this, because we used to have parallel processing logic here before and then decided to simplify during the review process and keep the TODO